packages/asset-service/src/generateAssetData/osmosis/index.ts

```diff
@@ -17,33 +18,72 @@ type OsmoAsset = {
   name: string
   display: string
   symbol: string
+  traces: {
+    type: string
+    counterparty: {
+      chain_name: string
+      bse_denom: string
+      channel_id: string
+    }
+    chain: {
+      channel_id: string
+    }
+  }[]
   logo_URIs: {
     png?: string
     svg?: string
   }
   coingecko_id: string
-  pools?: {
-    [key: string]: number
-  }
+  keywords: string[]
 }
 
 type OsmosisAssetList = {
-  chain_id: string
+  chain_name: string
   assets: OsmoAsset[]
 }
 
 export const getAssets = async (): Promise<Asset[]> => {
+  /** Helper function to get latest commit hash for default branch of given repo.
+   * This is used to construct githack URLs */
+  const getCurrentHash = memoize(async (repository: string): Promise<string> => {
```

Comment:
Curious why do we need this?
https://raw.githack.com/osmosis-labs/assetlists/main/osmosis-1/osmosis-1.assetlist.json works similarly, unless the default branch ever changes out of main upstream, which seems like a very unlikely scenario to me, we are good without a commit hash

Reply:
This serves two purposes. First, [GitHack FAQ](https://raw.githack.com/faq) instructs that we should be using CDN URLs (repo/hash/file) here, and we're violating the GitHack TOS by using development URLs (repo/branch/file) in a production application. Second, fetching and generating a CDN URL from the latest default branch commit hash ensures that the asset list is always up-to-date.

See more [here](https://raw.githack.com/).

Reply:
Still unsure about this.

According to the docs: ` To ensure that the CDN always serves the version of the file you want, use a git tag or commit ref in the file's path instead of a branch name, and update the URL if you push a new version of the file.`

Let's say you run the generation asset twice in a day and the latest upstream commit hash didn't change, the `master` blob for said file would still be the same.
Realistically, you'd only need to run this when the file actually changed, so using a commit hash or a branch name would produce the same effect.

Now the comment WRT using an actual CDN URL *is* valid:

<img width="1325" alt="image" src="https://user-images.githubusercontent.com/17035424/213907914-694c050c-e0bc-4398-bf27-0752a7e5d1bb.png">

However, we're still using `raw.githack.com` (dev) vs . `rawcdn.githack.com` (prod) here?

https://github.com/shapeshift/lib/blob/bb16508cbb7d05595d3df771f1707d9084aafdb5/packages/asset-service/src/generateAssetData/osmosis/index.ts#L73

Either way, given this is a run once, generate, forget and repeat a few weeks later kind of script, caching/rate-limiting is a non-issue for us.

If we want to add a safety factor, happy with changing to the prod URL, but I stand with the fact using a commithash is over-engineering for our use-case, let's not clutter the PR with it.

Reply:
@gomesalexandre Yeah, that's right. Using the branch name or the commit hash is functionally the same in that case. The reason I'm fetching the latest hash from Github instead of hardcoding a fixed commit hash is to be able to use CDN URLs and simultaneously make sure that we have the latest file contents.

Nice catch with the `raw.githack` vs r`awcdn.githack` typo! I'll fix that now.

I'm not worried about caching/rate limiting on our end with the asset list - it's the icon URLs that end up in `generatedAssetList.json` and baked into the production web app that I'm worried about. I did this to avoid any issues with clients being unable to fetch asset icons because of those URLs being rate-limited. Since I had that in place for the icons, I thought I might as well just slap it onto the asset list URL as well.